### ISSUE TITLE
remove --job-name from submit command line arguments

### ISSUE THREAD

**Issue by atravitz** [glotzerlab/signac-flow#164]
explicitly assigning a string to `--job-name` when submitting results in the status remaining at `[U]` even while the job is running. An example of code that causes this is:

```
hours=24
export HOOMD_WALLTIME_STOP=$((`date +%s` + $hours * 3600 - 10 * 60))
python project.py submit -o sample -w $hours -- --partition=gpu-shared --gres=gpu:1 --job-name=effective_potential
```

**Comment by vyasr** [comment]
I think that the proper place to handle this is by checking the contents of `flags` in `ComputeEnvironment.submit` and error out if `--job-name` is a valid option.

**Comment by bdice** [comment]
Context: signac-flow assigns a specific job name when submitting to the scheduler, so that it is possible to identify what jobs are running in the status check. That is, the name is a unique identifier that is used when querying the scheduler to determine whether an operation is queued/active/stopped.

**Comment by csadorf** [comment]
We could in principle record the custom job name, right? But is this something we want to support? I vote no.

**Comment by vyasr** [comment]
I also vote no. I think it's a reasonable limitation of flow that we use naming explicitly for our tracking. We could always allow overriding this with `--force`, with a warning to the user that our tracking logic won't work in that case.

**Comment by ac-optimus** [comment]
hi, can i work on this issue?

**Comment by ac-optimus** [comment]
I am not able to get the same error(status remainig [U]) as @atravitz. I am using this as a [example](https://github.com/vyasr/signac_scipy2018.git) for this issue. The status for each job submitted changes to  [A] while running and to [I] (status=3) after the execution. I am using ```SimpleSchedulerEnvironment``` as environment for my local system. Can you help me if there is some mistake in my practice or the discrepancy is due the environment used? Also I could notice that name of ```job_name``` does not change after setting flag ```--job-name``` (with and without --force) to something else.
Thanks!

**Comment by bdice** [comment]
@ac-optimus I am not sure if we can test this with the current SimpleScheduler. It does not allow the user to override the job name when submitting.

**Comment by vyasr** [comment]
Unfortunately @bdice is correct, you will not be able to test this very effectively with the SimpleScheduler. It does not really parse the job name argument. However, I think you should be able to test your changes anyway, because we want specifying a job name to fail before it even reaches the submission to the specific environment. The result of your change should be that we raise a `ValueError` or similar whenever the user specifies `--job-name`, so the submission should not succeed and it will never show up in the status output.

**Comment by ac-optimus** [comment]
Done that. Please review, but lets say if I want to reproduce the error then which environment should I use?

**Comment by vyasr** [comment]
Unfortunately, you picked one of the very few issues that I don't think we currently have any way of testing without a production environment available. The SimpleScheduler is very minimal and doesn't support enough of the features of a true Slurm or Torque scheduler to be able to capture this behavior. The problem is further compounded by the fact that the extra flags are passed directly to the call to the queuing command, so they won't show up in the printed job scripts. The only way to reproduce the bug is to set up a testing scheduler environment using something like [celery](http://www.celeryproject.org/) where you could actually submit a job with a different name and see that name show up.

**Comment by vyasr** [comment]
To clarify: we don't currently support celery, so you'd also have to add support for it. I was suggesting how one _could_ test this in principle, but it would require quite a bit of work to even get the environment set up. As long as you are able to test that the ValueError is raised when the job name is provided, I think you've done your part. The reviewers on the PR will be responsible for testing this on environments they have access to.

**Comment by ac-optimus** [comment]
@vyasr okays sure!

**Comment by vyasr** [comment]
Resolved by #241
